### ISSUE TITLE
Scrub synbiohub credentials from commit history

### ISSUE THREAD

**Issue by mwes** [SD2E/experimental-intent-parser#132]
In GitLab by @bbartley on Oct 16, 2019, 11:13

Doh!  I accidentally committed the synbiohub user credentials in one of my utility scripts.

**Comment by mwes** [comment]
In GitLab by @jakebeal on Jun 8, 2020, 14:12

assigned to @mweston and unassigned @bbartley

**Comment by mwes** [comment]
In GitLab by @jakebeal on Jun 8, 2020, 14:13

@mweston The commit in question should be indicated by the date of the issue.  After scrubbing, please move to GitHub.

**Comment by mwes** [comment]
In GitLab by @mweston on Jun 8, 2020, 14:19

@tramyn can you add your additional history request to this issue?

**Comment by mwes** [comment]
In GitLab by @tramyn on Jun 8, 2020, 14:23

yep. I would like this [credentials.json](https://gitlab.sd2e.org/sd2program/experimental-intent-parser/commit/f0b1419fff8065f2fb68aa88318babfc1cec3eeb) file removed from this repo's commit history.

**Comment by mwes** [comment]
In GitLab by @tramyn on Jun 12, 2020, 13:35

@mweston What is the current status of this? @jakebeal discussed moving intent parser from gitlab to github. I think this issue must be resolved before we can make that transition. 

If this issue is close to completion, @jakebeal do you have a specific github organization or account you want intent parser to exist on?

**Comment by mwes** [comment]
In GitLab by @mweston on Jun 12, 2020, 15:09

@tramyn sure I can complete this today. Note that this will forcefully re-write history on the repository. Do you have any pending work in local branches that is not checked in? Please ensure everything is pushed and let me know when you're ready. I'll test locally in the meantime.

**Comment by mwes** [comment]
In GitLab by @jakebeal on Jun 12, 2020, 15:49

This should move into the SD2 GitHub organization.

**Comment by mwes** [comment]
In GitLab by @tramyn on Jun 12, 2020, 16:42

@mweston Please go ahead and perform a history rewrite. I have everything I need on my end to prepare for this change.

**Comment by mwes** [comment]
In GitLab by @mweston on Jun 12, 2020, 19:15

@tramyn done. GitLab still has a cache, so if you know the commit hashes, it looks like you can still view them here, despite having re-written all of the commits.

The GitHub mirror, however, does not have those commits, and properly shows the credentials files and password being scrubbed:

https://github.com/SD2E/experimental-intent-parser/commit/23a7f30c82ca768a27dfd99999cf9533b02dfd04
https://github.com/SD2E/experimental-intent-parser/commit/0fb0d0ef120cd1a00558d07a7f80b713579f8e0b
https://github.com/SD2E/experimental-intent-parser/commit/786436900726440c74f76657c48a0afe700a4470

So this, and the mirroring dependency, should be resolved. I'll work with TACC to clean up the GitLab issue (it looks like there is some sort of rake task we can run to clear the Redis cache here: https://docs.gitlab.com/ee/user/project/repository/reducing_the_repo_size_using_git.html#repository-cleanup)

**Comment by mwes** [comment]
In GitLab by @mweston on Jun 15, 2020, 13:19

If you haven't already done so, @tramyn, you and other developers working on this repo should clone a fresh copy of the repository and resume work from there.

**Comment by mwes** [comment]
In GitLab by @tramyn on Jun 15, 2020, 14:46

Thank you @mweston. From what I'm seeing, you have ported intent parser over to GitHub. This is wonderful! Thank you for doing this. I assume that when you mention to clone a copy of the new repo, this mean clone a new copy of intent parser from gitlab and not GitHub. Correct?

Also, is there an option to move GitLab issues to GitHub as well or do I have to manually recreate each issue?

**Comment by mwes** [comment]
In GitLab by @mweston on Jun 15, 2020, 15:16

Yes - re-clone from GitLab. The commit ids have all been replaced.

I will look into the GL -> GH issue migration.

**Comment by mwes** [comment]
In GitLab by @tramyn on Jun 16, 2020, 13:56

@mweston, I do not have the ability to delete branches on experimental-intent-parser github repository. Is this limitation due to me being a member and not an owner of SD2E organization?

**Comment by mwes** [comment]
In GitLab by @mweston on Jun 16, 2020, 14:03

@tramyn I am looking - is the delete icon (trash icon) grayed out? Does it give an error?

**Comment by mwes** [comment]
In GitLab by @tramyn on Jun 16, 2020, 14:05

@mweston nvm, I had to browse to find it.

**Comment by mwes** [comment]
In GitLab by @mweston on Jun 16, 2020, 14:08

OK good - I confirmed you do have write access to the repository.

**Comment by jakebeal** [comment]
This has been completed.
